torchvision/prototype/datasets/_builtin/README.md

```diff
@@ -0,0 +1,127 @@
+
+## `resources(self, config)`
+
+Returns `List[datasets.utils.OnlineResource]` of all the files that need to be present locally before the dataset with a specific `config` can be build. The download will happen automatically. 
+
+Currently, the following `OnlineResource`'s are supported:
+
+- `HttpResource`: Used for files that are directly exposed through HTTP(s) and only requires the URL.
+- `GDriveResource`: Used for files that are hosted on GDrive and requires the GDrive ID as well as the `file_name`.
+- `ManualDownloadResource`: Used files are not publicly accessible and requires instructions how to download them manually. If the file does not exist, an error will be raised with the supplied instructions.
+
+Although optional in general, all resources used in the built-in datasets should comprise [SHA256](https://en.wikipedia.org/wiki/SHA-2) checksum for security. It will be automatically checked after the download. You can compute the checksum with system utilities or this snippet:
+
+```python
+import hashlib
+
+def sha256sum(path, chunk_size=1024 * 1024):
+    checksum = hashlib.sha256()
+    with open(path, "rb") as f:
+        for chunk in iter(lambda: f.read(chunk_size), b""):
+            checksum.update(chunk)
+    print(checksum.hexdigest())
+```
+
+### `_make_datapipe(resource_dps, *, config, decoder)`
+
+This method is the heart of the dataset that need to transform the raw data into a usable form. A major difference compared to the current stable datasets is that everything is performed through `IterDataPipe`'s. From the perspective of someone that is working with them rather than on them, `IterDataPipe`'s behave just as generators, i.e. you can't do anything with them besides iterating. 
+
+Of course, there are some common building blocks that should suffice in 95% of the cases. The most used 
+
+- `Mapper`: Apply a callable to every item in the datapipe. 
+- `Filter`: Keep only items that satisfy a condition.
+- `Demultiplexer`: Split a datapipe into multiple ones.
+- `IterKeyZipper`: Merge two datapipes into one.
+
+All of them can be imported `from torchdata.datapipes.iter`. In addition, use `functools.partial` in case a callable needs extra arguments.  If the provided `IterDataPipe`'s are not sufficient for the use case, it is also not complicated to add one. See the MNIST or CelebA datasets for example.
+
+`make_datapipe()` gets passes `resource_dps`, which is a list of datapipes that has a 1-to-1 correspondence with the return value of `resources()`. In case of archives with regular extension, the datapipe will contain tuples comprised of the path and the handle for every file in the archive. Otherwise the datapipe will only contain one of such tuples for the file specified by the resource.
```

Comment:
Potential typos:
- `gets passes` => `receives`
- `regular extension` => `regular expression`

Reply:
```suggestion
`make_datapipe()` receives `resource_dps`, which is a list of datapipes that has a 1-to-1 correspondence with the return value of `resources()`. In case of archives with regular suffixes (`.tar`, `.zip`, ...), the datapipe will contain tuples comprised of the path and the handle for every file in the archive. Otherwise the datapipe will only contain one of such tuples for the file specified by the resource.
```

Reply:
for the 2nd, I see what you meant. I thought you referred to the "regex style" paths you include in the implementation. The way you rewrote it makes it more clear. :)